cmd/immortal/main.go

```diff
@@ -49,11 +48,22 @@ func main() {
 	if len(cfg.Require) > 0 {
 		down := []string{}
 		ctl := &immortal.Controller{}
+		services, _ := ctl.FindServices(immortal.GetSdir())
+		if userServices, err := ctl.FindServices(
+			immortal.GetUserdir(),
+		); err == nil {
+			services = append(services, userServices...)
+		}
 		for _, r := range cfg.Require {
-			socket := filepath.Join(immortal.GetSdir(), r, "immortal.sock")
-			if status, err := ctl.GetStatus(socket); err != nil {
-				down = append(down, r)
-			} else if status.Up == "" {
+			isDown := true
```

Comment:
Here based on just the requirements could be better to search for the desired `services` rather than pre-filling `services` that may be small or huge depending on the current use case

Reply:
I was thinking because the search happens based on name and the folders in $HOME/.immortal are named after the `PID` we need to query all sockets to find out the actual `name` of the process?

Reply:
hi @loafoe writing to know if is working the latest commit as expected, in this specific snippet it will only search for sockets matching the required name instead of scanning all files.